makeflow/src/dag_file.c

```diff
@@ -84,4 +87,32 @@ int dag_file_in_trans( const struct dag_file *f )
 		return 0;
 }
 
+int dag_file_mount_clean(struct dag *d) {
+	struct list *list;
+	struct dag_file *df;
+	if(!d) return 0;
+
+	list = dag_input_files(d);
+	if(!list) return 0;
+
+	list_first_item(list);
+	while((df = (struct dag_file *)list_next_item(list))) {
+		if(df->source) {
```

Comment:
I think line 100-107 are what should remain in dag_file.c. Then a function "dag_mount_clean" could iterate over the the files calling dag_file_mount_clean as well as cleaning the dag level information such as cache_dir.